tests/unit/test_cli.py

```diff
@@ -189,6 +191,40 @@ def side_effect():
         mock_daemon_start.assert_called_once()
 
 
+    def test_ensure_directory_exists_not_exists(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            test_dir = pathlib.Path(tempdir).joinpath("test_ensure")
+            cli.ensure_directory_exists(str(test_dir))
+            self.assertTrue(test_dir.is_dir())
+
+
+    def test_ensure_directory_exists_not_writable(self):
+        if os.getuid() == 0:
+            logger = logging.getLogger(__name__)
+            logger.warning("test_ensure_directory_exists_not_writable cannot fail when run as root")
+            # prefer un-failable test to un-passable test
+            return
```

Comment:
I actually added this because I encountered the case. To avoid juggling multiple python installations on my machine, I did my testing by mounting my repository as a volume in a docker container. Since it was just a personal development container, I didn't bother adding a user, and so the tests ended up running as root.

The test not passing stumped me for a good 15 minutes, and may be confusing to others that test the same way. I can change this to fail on root, but I would still recommend detecting the case and logging a warning.

Reply:
The issue is that we do not take something like this into account when writing other tests. Perhaps other tests won't be as sensitive to such a scenario but we don't know for sure and nobody takes this into account as far as I know.

I think it's safe to assume that the tests are always run as regular users (not as root) because the resultant `lbrynet` binary is also not intended to run as root nor the wallet server.

Reply:
Ok, I'll remove the check.